File: src/main/webapp/src/model/request/ChangePasswordRequest.ts
/**
 * ChangePasswordRequest
 * @author Simpli CLI generator
 */
import {$} from '@/facade'
import {Request, ResponseExclude} from '@simpli/serialized-request'

/* TODO: review generated class */
export class ChangePasswordRequest {
  @ResponseExclude()
  currentPassword: string | null = null

  @ResponseExclude()
  newPassword: string | null = null

  @ResponseExclude()
  confirmPassword: string | null = null

  /**
   * Changes the password with a given new password
   */
  async changePassword() {
    const request = $.utils.clone(this)
    request.currentPassword = $.utils.sha256(this.currentPassword)
    request.newPassword = $.utils.sha256(this.newPassword)
    request.confirmPassword = $.utils.sha256(this.confirmPassword)

    return await Request.post(`/client/auth/me/password`, request)
      .name('changePassword')
      .asNumber()
      .getData()
  }
}

File: src/main/kotlin/org/daonators/client/request/ChangePasswordRequest.kt
package org.daonators.client.request

import org.daonators.exception.response.BadRequestException
import br.com.simpli.model.LanguageHolder

/**
 * Change Password Request Model
 * @author Simpli CLI generator
 */
class ChangePasswordRequest(var currentPassword: String?, var newPassword: String?, var confirmPassword: String?) {
    fun validate(lang: LanguageHolder) {
        if (currentPassword.isNullOrEmpty()) {
            throw BadRequestException(lang.cannotBeNull("Current Password"))
        }

        if (newPassword.isNullOrEmpty()) {
            throw BadRequestException(lang.cannotBeNull("New Password"))
        }

        if (confirmPassword.isNullOrEmpty()) {
            throw BadRequestException(lang.cannotBeNull("Confirm Password"))
        }

        if (newPassword != confirmPassword) {
            throw BadRequestException(lang["password_no_match"])
        }
    }
}
